quickcheck/Test/Tasty/QuickCheck.hs

```diff
@@ -82,7 +82,10 @@ testProperties name = testGroup name . map (uncurry testProperty)
 newtype QuickCheckTests = QuickCheckTests Int
   deriving (Num, Ord, Eq, Real, Enum, Integral, Typeable)
 
-newtype QuickCheckReplay = QuickCheckReplay (Maybe Int)
+-- | Replay seed. The @Left int@ form is kept for legacy purposes.
+-- The form @Right (qcgen, intSize)@ holds both the seed and the size
+-- to run QuickCheck tests.
+newtype QuickCheckReplay = QuickCheckReplay (Maybe (Either Int (QCGen, Int)))
```

Comment:
Could it be
```suggestion
newtype QuickCheckReplay = QuickCheckReplay (Maybe (Int, Maybe QCGen))
```
?

Reply:
I find the code easier to approach as it is.

* The two Ints in the Either type have different meanings, whether the meaning of the Int in the Maybe form depends on the value of the second component of the pair.

* The quickcheck API can take the pair `(QCGen, Int)` as is without further ado.

But let me know if you still think the `Maybe` form is to be preferred.

Another option is to use a custom type:
```haskell
data Seed = OldSeed Int | NewSeed (QCGen, Int)
```

Reply:
OK, unless someone else feels strongly about it, I'm fine to keep it as is.